### ISSUE TITLE
Question: fries-module

### ISSUE THREAD

**Issue by madpat3** [acemod/ACE3#4072]
i made a mod-helicopter ace-fries compatible. i put it on map and synced it with the frie-modul. fine so far.
but when the chopper gets destroyed and respawns, it respawns without fries. is there a way to get around this? like putting some lines into its init?

**Comment by commy2** [comment]
Vehicles cannot respawn. The BI respawn module creates a new vehicle and that one was obviously never synched to the module. It's the same problem you have with spawned in helicopters (e.g. zeus or another script).

I'd use this personally:
```
["classname_helicopter", "initPost", ace_fastroping_fnc_equipFRIES, nil, nil, true] call CBA_fnc_addClassEventHandler;
```
Not tested.

**Comment by madpat3** [comment]
you are a Genius!
and is there a way to do that with a specific vehicle, so that not every vehicle of that class has fries-capability?

**Comment by commy2** [comment]
I don't think so, no.

What you can do is to check at which position they spawn, and only execute the equipFries function if they spawn at the place, where all fries equipped helicopters should spawn.

```
MyMission_FriesSpawnPosition = [<x,y,z>];

["<classname_helicopter>", "initPost", {
    params ["_vehicle"];

    if (getPosASL _vehicle distance MyMission_FriesSpawnPosition  < 10) then {
        _vehicle call ace_fastroping_fnc_equipFRIES;
    };
}, nil, nil, true] call CBA_fnc_addClassEventHandler;
```

**Comment by madpat3** [comment]
i put in the coordinates x,y,z of the position, where the heli respawns. and put in the class Name of the heli. but it ain't working. or do you mean the Position of the respawn-module itself?

is the Syntax right, with all the gaps in between?

MyMission_FriesSpawnPosition = [14226.381,16201.092,-0.006];

["bw_ec135_transport", "initPost", {
    params ["_vehicle"];

    if (getPosASL _vehicle distance MyMission_FriesSpawnPosition  < 10) then {
        _vehicle call ace_fastroping_fnc_equipFRIES;
    };
}, nil, nil, true] call CBA_fnc_addClassEventHandler;

**Comment by commy2** [comment]
position of the spawn point. If it doesn't work, then I guess it's the way the respawn module spawns these things (probably at 0,0,0).

I don't think this can be solved. The problem is that the "respawning" vehicles have nothing to do with the old ones.

**Comment by madpat3** [comment]
nevertheless, it works with classes now and it works with zeus. 
which means, you could use it, to introduce a FRIES-Modul, which can be set by zeus, right?

**Comment by markooff** [comment]
Hello :) 
U've just tried using of the method  with the 
["<classname_helicopter>", "initPost", ace_fastroping_fnc_equipFRIES, nil, nil, true] call CBA_fnc_addClassEventHandler;
in my init.sqf file , but I've got a strange effect 
- after my helicopter (precisely - it was RHS UH1Y Unarmed - RHS_UH1Y_UNARMED ) respawned - it has visible FRIES mechanism , but You cannot use it anymore... 
It looks like there were some minor changes in CBA/ACE invocation of that module during last few months (or years) , but I havent found anything about it in Framework documentation Wiki , or user documentation as well .

Or, maybe , am I wrong ?

**Comment by markooff** [comment]
In addition - info : 
The adding of FRIES module works only before the helicopter is deleted - because it has a fraze :
[this] call ace_fastroping_fnc_equipFRIES; 
in his init field - and it works well (but only once). After the helicopter respawn - the situation is as I described above ... I think about some stupid errors in the invocations of CBA_fnc_addClassEventHandler - but there are any of them, because I've checked this invocation in CBA github exapmles - and it is correct. 
So the question is - why the FRIES module behaves that way ?

**Comment by BaerMitUmlaut** [comment]
Please create a new issue if you have a problem. Nobody looks into closed issues

**Comment by Lamp-master** [comment]
I try just put in 
call ace_fastroping_fnc_equipFRIES;
this code  expression in BI_vehiclerespawnModule
then, When UH60 respawned, they respawned FRIES system on.

**Comment by BaerMitUmlaut** [comment]
Buddy, this is an issue from 2016. Please read my last comment.
